git_downloader.py

```diff
@@ -16,53 +17,91 @@ def main(args, loglevel):
     logging.basicConfig(format="%(levelname)s: %(message)s", level=loglevel)
 
     socket.setdefaulttimeout(args.timeout)
-    
-    g = Github()
-    with open(args.repo_file, 'r') as f:
-        file_counter = 0
-        for line in f.readlines():
-            logging.info('Fetching repository: %s' % line)
-            try:
-                repo_str = line.rstrip().split('github.com/')[-1]
-                repo = g.get_repo(repo_str)
-                tree = repo.get_git_tree('master', recursive=True)
-                files_to_download = []
-                for file in tree.tree:
-                    if fnmatch.fnmatch(file.path, args.wildcard):
-                        files_to_download.append('https://github.com/%s/raw/master/%s' % (repo_str, file.path))
-                for file in files_to_download:
-                    logging.info('Downloading %s' % file)
-                    file_counter += 1
-                    filename = posixpath.basename(urlparse.urlsplit(file).path)
-                    output_path = os.path.join(args.output_dir, filename)
-                    if os.path.exists(output_path):
-                        output_path += "-" + str(file_counter)
-                    try:
-                        urlretrieve(file, output_path)
-                    except Exception as e:
-                        logging.exception('Error downloading %s.' % file)
-            except Exception as e:
-                 logging.exception('Error fetching repository %s.' % line)
+
+    if args.username and args.password:
+      g = github.Github(args.username, args.password)
+    else:
+      g = github.Github()
+
+    if args.repo_file:
+        repo_gen = file_repo_gen(args.repo_file, g)
+    else:
+        repo_gen = global_repo_gen(args, g)
+    download_files(args, g, repo_gen)
+
+def file_repo_gen(repo_file, g):
+    with open(repo_file, 'r') as f:
+        for line in f:
+            repo_str = line.rstrip().split('github.com/')[-1]
+            yield g.get_repo(repo_str)
+
+
+def global_repo_gen(args, g):
+    git_repo_gen = g.get_repos(since=args.last_repo)
+    for repo in git_repo_gen:
+        yield repo
+
+
+def download_files(args, g, repo_gen):
+    file_counter = 0
+    for repo in repo_gen:
+        logging.info('Fetching repository: %s' % repo.full_name)
```

Comment:
Are you sure you don't want to print the repository id as well? In case you do, you could use the last printed number as the `last_repo` script argument